### PR TITLE
Additional features: specific comment deletion and timestamp sorting

### PR DESCRIPTION
**Pull request by tatiana-s**
The comment section now additionally 

- provides the option to enter a username and a mood for each comment
- supports deletion of specific comments (instead of all comments at once like before)
- can be displayed in different sort orders (newest or oldest first).

Screenshot 2020-07-17 at 7 23 55 PM

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/data/Comment.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.data;
+
+/** A comment. */
```
Comment (nataliakrylova) [inline portfolio/src/main/java/com/google/sps/data/Comment.java]: (nit) Maybe add a bit longer description about what this is? :)

File: portfolio/src/main/java/com/google/sps/servlets/CommentServlet.java [hunk 2]
```diff
@@ -34,43 +37,83 @@
...
+  private static final String SORT_ORDER_QUERY_PARAM = "sort";
+  private static final String DEFAULT_USERNAME = "Anonymous";
 
   /** Loads and returns comments from the datastore database. */
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    int commentNumber = Integer.parseInt(request.getParameter(COMMENT_NUMBER_QUERY_PARAM));
     Query query = new Query(CommentEntity.KIND.getLabel());
+    String sortOrder = request.getParameter(SORT_ORDER_QUERY_PARAM);
+    switch (sortOrder) {
```
Comment (nataliakrylova) [inline portfolio/src/main/java/com/google/sps/servlets/CommentServlet.java:54]: case "old":
...
case "new":
default:
...

Unless new and default will be different in the future.

File: portfolio/src/main/webapp/script.js [hunk 3]
```diff
@@ -14,8 +14,13 @@
 
 /* eslint-disable no-unused-vars */
 
+/** Current slide index in photo gallery. */
 let slideIndex = 0;
 
+/** Current filter settings in  comment section. */
```
Comment (nataliakrylova) [inline portfolio/src/main/webapp/script.js]: (supernit) double space between in and comment :)

File: portfolio/src/main/webapp/style.css [hunk 4]
```diff
@@ -82,24 +88,52 @@ input[type=submit] {
...
-  color: white;
-  padding: 5px 20px;
-  text-align: center;
+  margin: 5px;
+  max-width: 100%;
+  padding: 5px 5px;
+  width: 100px;
+}
+
+#select-mood {
```
Comment (nataliakrylova) [inline portfolio/src/main/webapp/style.css:130]: The box seems to cut emoji from the bottom, maybe changing its size or can be helpful?

File: portfolio/src/main/webapp/script.js [hunk 5]
```diff
@@ -49,10 +54,26 @@ function showSlides(newIndex) {
...
+/** Called when comment number limit selector changes. */
+function changeLimit() {
+  const limit = document.getElementById('select-comment-number').value;
+  commentLimit = limit;
+  showComments();
+}
+
+/** Called when sorting order selector changes. */
+function changeSort() {
+  const sort = document.getElementById('select-comment-sort').value;
```
Comment (nataliakrylova) [inline portfolio/src/main/webapp/script.js]: Why not sortOrder = document.... ?

File: portfolio/src/main/webapp/script.js [hunk 6]
```diff
@@ -49,10 +54,26 @@ function showSlides(newIndex) {
...
 }
 
+/** Called when comment number limit selector changes. */
+function changeLimit() {
+  const limit = document.getElementById('select-comment-number').value;
+  commentLimit = limit;
+  showComments();
+}
+
+/** Called when sorting order selector changes. */
```
Comment (nataliakrylova) [inline portfolio/src/main/webapp/script.js]: You should explain what the function does.- so in this case changes sorting order.

**Review by nataliakrylova — COMMENTED** [review]
Overall UX comment: when user leaves a comment there is no feedback regarding whether the comment wasn't added because of an error or because something is missing, it's not clear which fields are required (usually marked with red star). Do you think you could make some improvements related to that and add maybe an error message, if possible?

File: portfolio/src/main/webapp/style.css (hunk 4 above)
Comment (tatiana-s) [reply portfolio/src/main/webapp/style.css:130]: Interesting, it doesn't cut it for me? I've set height now so maybe that will help but I can't see myself.

**Comment by tatiana-s** [comment]
Will probably go back to adding an error message if comments aren't added later if I have the time but leave it as it is for now.

File: portfolio/src/main/webapp/script.js [hunk 7]
```diff
@@ -54,17 +54,20 @@ function showSlides(newIndex) {
...
+/** Called when comment number limit selector changes: changes limit. */
 function changeLimit() {
-  const limit = document.getElementById('select-comment-number').value;
-  commentLimit = limit;
+  commentLimit = document.getElementById('select-comment-number').value;
   showComments();
 }
 
-/** Called when sorting order selector changes. */
+/** Called when sorting order selector changes an: changes sorting order. */
```
Comment (nataliakrylova) [inline portfolio/src/main/webapp/script.js]: an: = and? :)

File: portfolio/src/main/webapp/style.css (hunk 4 above)
Comment (nataliakrylova) [reply portfolio/src/main/webapp/style.css:130]: Hm, strange... It looks the same but it's not too bad, so let's keep it like this :)

**Review by nataliakrylova — APPROVED** [review]

**Pull request merged**